### DISCUSSION TITLE
Does YOLOv8 artificially increase the resolution on low-res training/validation images?

### DISCUSSION THREAD

**Discussion by mytakara** [ultralytics/ultralytics#8307] [Q&A]
I'm working on a custom detection model that trains/validates on low resolution images. However, when I view the sample train/val images, they seem to be significantly upsampled compared to the original input images. See the examples below: I used extremely low resolution just to make it easier to see the difference. 

The top images are the originals at 48p annotated with labels and the bottom are the same images annotated with predicted boxes, but look like the resolution has been doubled.

<img width="1107" alt="yolo_labels" src="https://github.com/ultralytics/ultralytics/assets/98113678/d186a359-e54e-4bcc-80b9-6e4e24b65cd9">
<img width="1108" alt="yolo_predictions" src="https://github.com/ultralytics/ultralytics/assets/98113678/968e4dc5-e75d-420c-8a67-4178a9415461">

I'm using a pretty basic approach (I think):

`MODEL = "yolov8n.pt"`
`model = YOLO(MODEL)`
`model.train(data="config.yaml", epochs=num_epochs)`

Everything else seems to be working as it should, but I'm confused by why the images with predicted boxes are much higher resolution. I want to make sure this isn't somehow skewing the performance of my fine-tuned model - it seems to perform _too_ well on the 48p images. Is this expected to happen? Is there a minimum resolution that YOLOv8 can handle? Could someone explain what might be causing this?

**Comment by pderrenger** [comment]
Hey there! 👋

YOLOv8 does resize images during training and validation to the `imgsz` specified in your training configuration or the default size if not specified. This is done to ensure consistency in input size across all images, which is crucial for the model to learn effectively. 

For your low-resolution images, if they appear to be upsampled, it's likely because they're being resized to match the `imgsz`. This doesn't artificially enhance the image quality; it simply scales them up to a uniform size. Here's a snippet to specify the image size during training, which should match your low-res input for clarity:

```python
model.train(data="config.yaml", epochs=num_epochs, imgsz=48)
```

If you're seeing unexpectedly high performance, it's worth double-checking your validation dataset and metrics to ensure they're aligned with your training setup. YOLOv8 can handle low-res images, but performance typically correlates with the resolution of the training data.

Hope this clears things up! 😊 Let me know if you have any more questions.

**Reply by mytakara** [reply]
Thank you for your response! 
To clarify, you are saying that images are rescaled to a uniform size before going through the train/val loops. 
I added the imgsz argument and the low resolution images have lower mAP@50to95 performance, which is what I expected, so it seems to be working as it should.

**Reply by pderrenger** [reply]
Hi there!

You've got it exactly right! 🎯 The images are indeed rescaled to a uniform size specified by the `imgsz` argument before training and validation. I'm glad to hear that adjusting the `imgsz` gave you the expected results with the mAP @mytakara performance. It sounds like everything is on track now. If you have any more questions or need further assistance, feel free to reach out. Happy training! 😊🚀
